hail/src/main/scala/is/hail/expr/types/physical/PCanonicalArray.scala

```diff
@@ -104,19 +104,26 @@ final case class PCanonicalArray(elementType: PType, required: Boolean = false)
     !isElementDefined(aoff, i)
 
   def setElementMissing(aoff: Long, i: Int) {
-    assert(!elementType.required)
-    Region.setBit(aoff + lengthHeaderBytes, i)
+    if (!elementType.required)
+      Region.setBit(aoff + lengthHeaderBytes, i)
   }
 
   def setElementMissing(aoff: Code[Long], i: Code[Int]): Code[Unit] =
-    Region.setBit(aoff + lengthHeaderBytes, i.toL)
+    if (elementType.required)
+      Code._empty
```

Comment:
should the fatal be here, not on srvb?

Reply:
alternatively, could `assert(!elementType.required)`?

Reply:
Good, yeah, I moved the assert.  I also did the same thing for PBaseStruct.